include/ignition/math/Interval.hh

```diff
@@ -0,0 +1,292 @@
+        {
+          if (this->rightValue <= _other.rightValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (this->rightValue < _other.rightValue)
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      /// \brief Check if the interval intersects `_other` interval
+      /// \param[in] _other interval to check for intersection
+      /// \return true if both intervals intersect, false otherwise
+      public: bool Intersects(const Interval<T> &_other) const
+      {
+        if (this->Empty() || _other.Empty())
+        {
+          return false;
+        }
+        if (this->rightClosed && _other.leftClosed)
+        {
+          if (this->rightValue < _other.leftValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (this->rightValue <= _other.leftValue)
+          {
+            return false;
+          }
+        }
+        if (_other.rightClosed && this->leftClosed)
+        {
+          if (_other.rightValue < this->leftValue)
+          {
+            return false;
+          }
+        }
+        else
+        {
+          if (_other.rightValue <= this->leftValue)
+          {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      /// \brief Equality test operator
+      /// \param _other interval to check for equality
+      /// \return true if intervals are equal, false otherwise
+      public: bool operator==(const Interval<T> &_other) const
+      {
+        return this->Contains(_other) && _other.Contains(*this);
+      }
+
+      /// \brief Inequality test operator
+      /// \param _other interval to check for inequality
+      /// \return true if intervals are unequal, false otherwise
+      public: bool operator!=(const Interval<T> &_other) const
+      {
+        return !this->Contains(_other) || !_other.Contains(*this);
+      }
+
+      /// \brief Stream insertion operator
+      /// \param _out output stream
+      /// \param _interval Interval to output
+      /// \return the stream
+      public: friend std::ostream &operator<<(
+        std::ostream &_out, const ignition::math::Interval<T> &_interval)
+      {
+        return _out << (_interval.leftClosed ? "[" : "(")
+                    << _interval.leftValue << ", " << _interval.rightValue
+                    << (_interval.rightClosed ? "]" : ")");
+      }
+
+      /// \brief The leftmost interval value
+      private: T leftValue{0};
+      /// \brief The righmost interval value
+      private: T rightValue{0};
+      /// \brief Whether the interval is left-closed or not
+      private: bool leftClosed{false};
+      /// \brief Whether the interval is right-closed or not
+      private: bool rightClosed{false};
+    };
+
+    namespace detail {
+       template<typename T>
+       const Interval<T> gUnboundedInterval =
```

Comment:
there may be some issues with static initialization of this variable (see #269). It looks like this is almost following the pattern from #283, but it is not a `constexpr`, so I think some additional work will be needed